staging/src/k8s.io/component-base/config/types.go

```diff
@@ -62,6 +62,12 @@ type LeaderElectionConfiguration struct {
 	// resourceLock indicates the resource object type that will be used to lock
 	// during leader election cycles.
 	ResourceLock string
+	// resourceName indicates the name of resource object that will be used to lock
+	// during leader election cycles.
+	ResourceName string
+	// resourceName indicates the namespace of resource object that will be used to lock
+	// during leader election cycles.
+	ResourceNamespace string
```

Comment:
adding these fields (and corresponding flags) makes the scheduler configuration (which has its own lockObjectName/Namespace fields already) ambiguous. As part of adding these, we have to resolve that ambiguity in the scheduler config.

We could do the following in the scheduler config:
1. change the scheduler [lock-object-name/namespace flags to bind to the new struct fields](https://github.com/kubernetes/kubernetes/blob/master/cmd/kube-scheduler/app/options/deprecated.go#L61-L62), mark the flags as deprecated in favor of the new flags
2. remove the [LockObjectName/LockObjectNamespace fields from the internal scheduler configuration type](https://github.com/kubernetes/kubernetes/blob/master/pkg/scheduler/apis/config/types.go#L141-L144) (leave the fields in the external configuration type for compatibility)
3. mark the [LockObjectName/LockObjectNamespace fields in the external scheduler configuration type](https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/kube-scheduler/config/v1alpha1/types.go#L137-L140) as deprecated in favor of the new standard fields
4. modify the [scheduler configuration defaulting](https://github.com/kubernetes/kubernetes/blob/master/pkg/scheduler/apis/config/v1alpha1/defaults.go#L81-L86) to only default the LockObjectName/LockObjectNamespace fields if the new fields are not set
5. modify the [scheduler configuration conversion](https://github.com/kubernetes/kubernetes/blob/master/pkg/scheduler/apis/config/v1alpha1/zz_generated.conversion.go#L198-L224) to implement custom conversion methods to do the following:
    * Convert_v1alpha1_KubeSchedulerConfiguration_To_config_KubeSchedulerConfiguration
      * call autoConvert_v1alpha1_KubeSchedulerConfiguration_To_config_KubeSchedulerConfiguration, then:
      * If only one of LockObjectName/ResourceName is set (or if they are both set and match), copy that to internal ResourceName
      * If only one of LockObjectNamespace/ResourceNamespace is set (or if they are both set and match), copy that to internal ResourceNamespace
      * If old/new fields are both set and do not match, error
    * Convert_config_KubeSchedulerConfiguration_To_v1alpha1_KubeSchedulerConfiguration:
      * call autoConvert_config_KubeSchedulerConfiguration_To_v1alpha1_KubeSchedulerConfiguration, then:
      * Copy internal ResourceName to external LockObjectName and ResourceName
      * Copy internal ResourceNamespace to external LockObjectNamespace and ResourceNamespace

Reply:
Addressed the comment.